src/repr/src/adt/interval.rs

```diff
@@ -41,12 +41,14 @@ impl Default for Interval {
     }
 }
 
-impl std::ops::Neg for Interval {
-    type Output = Self;
-    fn neg(self) -> Self {
-        Self {
-            months: -self.months,
-            duration: -self.duration,
+impl num_traits::ops::checked::CheckedNeg for Interval {
+    fn checked_neg(&self) -> Option<Self> {
+        if let (Some(months), Some(duration)) =
+            (self.months.checked_neg(), self.duration.checked_neg())
```

Comment:
Because the MIN/MAX values for duration are not equal w/ an `abs` operation,  this ought to go through `repr::adt::Interval::new()`, i.e.

here's the max:

```
SELECT INTERVAL '2147483647 days 2147483647 hours 59 minutes 59.999999 seconds';
            interval             
---------------------------------
 2236962132 days 07:59:59.999999
```

here's the min:
```
materialize=> SELECT INTERVAL '-2147483648 days -2147483648 hours -59 minutes -59.999999 seconds';
             interval              
-----------------------------------
 -2236962133 days -08:59:59.999999
```

here's the negated min:

```
SELECT -INTERVAL '-2147483648 days -2147483648 hours -59 minutes -59.999999 seconds';
            ?column?             
---------------------------------
 2236962133 days 08:59:59.999999
```

2236962133 days from the negated min exceeds the 2236962132 days from the max.

This should likely be a test, as well.

Reply:
Good catch, just added the fix and test.